### PR TITLE
Generalisation

### PR DESCRIPTION
**Pull request by Laudut**
plot a matrix of maps corresponding to lists of SIZE_FACTOR and VALUE_FACTOR

### CODE REVIEW

File: perlin.py [hunk 1]
```diff
@@ -79,62 +80,57 @@ def setWaterLevel(map, seaLevel=0):
...
-axs[1].imshow(colormap1a)
-axs[1].set_title("SECOND_GRID_SIZE x SECOND_SIZE_FACTOR = " + str((SECOND_GRID_SIZEa, SECOND_SIZE_FACTORa)))
+for i in range(N):
+    GRIDS[i] = randomGradGrid2D([GRID_SIZE[i]+1, GRID_SIZE[i]+1])
+    MAPS[i] = generateMap2D(GRIDS[i], SIZE_FACTOR[i])
+    for j in range(M):
+        CMAPS[j][i] = BASE_VALUE_FACTOR[j] * np.array(MAPS[0]) + SECOND_VALUE_FACTOR[j] * np.array(MAPS[i])
+        enhmap[j][i], colormap[j][i] = setWaterLevel(CMAPS[j][i], SEA_LEVEL) 
+    
+        axs[j,i].imshow(colormap[j][i])
```
Comment (ebrossier-secher) [inline perlin.py]: Just a small thing to fix : When we use a single BASE_VALUE_FACTOR, M = 1 and thus, `axs[j, i]` doesn't exist, it should be `axs[i]`.

The rest is perfect !

**Review by ZynoXelek — CHANGES_REQUESTED** [review]

File: perlin.py (hunk 1 above)
Comment (Laudut) [reply perlin.py]: You are right !
If fix it with a conditional if, and try it : it seems to work well
I let you resolve this conversation.

Comment (ebrossier-secher) [reply perlin.py]: Yep, perfect !

**Review by ZynoXelek — APPROVED** [review]
It's all good for me !

**Pull request merged**